packages/angular_devkit/schematics/src/formats/html-selector.ts

```diff
@@ -5,18 +5,47 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-
 import { schema } from '@angular-devkit/core';
 
+// As per https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name
+// * Without mandatory `-` as the application prefix will generally cover its inclusion
+// * And an allowance for upper alpha characters
+
+// NOTE: This should eventually be broken out into two formats: full and partial (allows for prefix)
+
+const unicodeRanges = [
+  [0xC0, 0xD6],
+  [0xD8, 0xF6],
+  [0xF8, 0x37D],
+  [0x37F, 0x1FFF],
+  [0x200C, 0x200D],
+  [0x203F, 0x2040],
+  [0x2070, 0x218F],
+  [0x2C00, 0x2FEF],
+  [0x3001, 0xD7FF],
+  [0xF900, 0xFDCF],
+  [0xFDF0, 0xFFFD],
+  [0x10000, 0xEFFFF],
+];
+
+function isValidElementName(name: string) {
```

Comment:
Seeing this https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name

Valid elements should not contain upper cases.

Nit: add return type.

Reply:
I made an exception for that since this is currently used for directives as well and many/most of them are camel case. The comment block above has a note about the inclusion of upper alpha characters.
I also didn't want to at least initially cause any further restrictions on the naming.  There's also a list of reserved element names that is also not included but since this is also used for partial (no prefix) names, checking for them doesn't work well currently.
But I agree that we should eventually be more restrictive for the actual element names.  This would require changes to the schematics as well as additional rules for custom attribute names.

Reply:
In that case LGTM.